### PR TITLE
Support for pages, api keys, and rendering

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+}
+```
+
+## Retrieving the home page and its children
+
+Now your app can access:
+
+`/api/v1/pages`
+
+To get information about the home page and its children. The response isa single JSON object with `slug`, `path`, `title`, `type`, `_url` and other properties describing the home page, similar to the way pieces are returned (see the "products" examples above). In addition, information about children of the home page is returned.
```
Comment (boutell) [inline README.md]: Typo isa

File: README.md [hunk 2]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+
+Armed with the `_id`, you can obtain detailed information about a page by making a separate API request:
+
+`/api/v1/pages/ID_GOES_HERE`
+
+A page returned in this way may in turn offer its own `_children` property.
+
+This response will include schema fields, areas, etc. in the same detail as it would when requesting a piece.
+
+*The `_children` property does not necessarily exist if there are no child pages.*
```
Comment (boutell) [inline README.md]: This is lame, make sure _children exists, even if empty, less work for consumer.

File: README.md [hunk 3]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+
+### Accessing ancestor pages
+
+Pages other than the home page will also have an `_ancestors` array. This functions similarly to the `_children` array.
```
Comment (boutell) [inline README.md]: The first element is the home page, the last is the parent.

File: README.md [hunk 4]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+
+### Nested tree response
+
+The response will be a single object representing the home page, with at least `title`, `slug`, `tags`, `_url` and `_id` properties, and a `_children` array. For speed, the response will not be as detailed as in a regular request to `/api/v1/pages`.
+
+The pages in the `_children` array, in turn, will feature their own `_children` arrays where needed, with a similarly limited level of detail.
+
+### Flat response
+
+It is possible to obtain a flat version of this data by adding `?flat=1` to the URL. In this case, a flat JSON array is returned. The array is sorted by depth, then by rank. Pages may still have a `_children` array, however it will only contain the `_id`s of the child pages, not the pages themselves. In this way you can still reconstruct the tree if you wish.
```
Comment (boutell) [inline README.md]: &flat=1

File: README.md [hunk 5]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+
+The body of your POST should contain all of the schema fields you wish to set, **and in addition it must contain a `_parentId` property** (note the underscore). The page will be added as the last child of the specified parent page.
+
+**The use of a JSON body, rather than traditional URL encoding, is strongly recommended and if you are working with areas it is mandatory.**
+
+On success you will receive a 200 status code and a JSON object containing the new page.
+
+## Updating a page
+
+To update a product, make a PUT request. Send it to:
```
Comment (boutell) [inline README.md]: Page not produc t

File: README.md [hunk 6]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+
+You may use either traditional URL-style encoding or a JSON body. **However if you are working with Apostrophe areas you must use a JSON body** (see below).
+
+**You may not move a page in the page tree via this method. The `path`, `level` and `rank` properties cannot be modified by this method.** To move a page in the page tree, see ["moving a page in the page tree,"](#moving-a-page-in-the-page-tree) below.
+
+## Deleting a page
+
+To delete a page, make a DELETE request. Send it to:
+
+`/api/v1/products/cxxxxxxx`
```
Comment (boutell) [inline README.md]: pages not products

File: README.md [hunk 7]
```diff
@@ -367,3 +445,122 @@ You will receive a JSON object in response. Using the `_id` property, you can cr
...
+
+To move a page in the page tree, make a POST request to the following URL:
+
+/api/v1/pages/move
+
+Your POST body must contain the following fields:
+
+* `_id` must be the `_id` of the page to be moved.
+* `relatedId` must be the _id of another page.
+* `relationship` must be `before`, `after` or `inside`. The page referenced by `_id` is moved `before`, `after` or `inside` the page specified by `relatedId`. If `_inside` is specified, the page becomes the last child of `relatedId`.
```
Comment (boutell) [inline README.md]: what's with the runaway italics?

File: lib/modules/apostrophe-pages-headless/index.js [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+          }
+          return res.send({});
+        });
+      });
+
+      // Move page
+      self.apos.app.post(endpoint + '/:id/move', function(req, res) {
+        var id = self.apos.launder.id(req.params.id);
+        var targetId = self.apos.launder.id(req.body.targetId);
+        var position = self.apos.launder.id(req.body.position);
```
Comment (stuartromanek) [inline lib/modules/apostrophe-pages-headless/index.js]: launder as string

**Pull request merged**